user-console/src/main/java/org/pentaho/mantle/client/ui/BurgerMenuBar.java

```diff
@@ -0,0 +1,320 @@
+      // Add a back menu item, followed by a menu item separator.
+      insertSeparator( new PentahoMenuSeparator(), 0 );
+      insertItem( new BurgerBackMenuItem( parentMenuItem.getHTML(), true ), 0 );
+    }
+
+    for ( MenuItem menuItem : getItems() ) {
+      BurgerMenuBar subMenu = (BurgerMenuBar) menuItem.getSubMenu();
+      if ( subMenu != null ) {
+        subMenu.addBackItems( menuItem );
+      }
+    }
+  }
+
+  private boolean hasItemsButNoBackItem() {
+    List<MenuItem> menuItems = getItems();
+    if ( !menuItems.isEmpty() ) {
+      return !( menuItems.get( 0 ) instanceof BurgerBackMenuItem );
+    } else {
+      return false;
+    }
+  }
+  // endregion Back Menu Items
+
+  protected void closeRoot() {
+    PopupPanel menuPopup = getRootPopup();
+    if ( menuPopup != null ) {
+      menuPopup.hide();
+    }
+  }
+
+  protected BurgerMenuBar getRootMenu() {
+    BurgerMenuBar parentMenu = getMenuBarParent();
+    if ( parentMenu != null ) {
+      return parentMenu.getRootMenu();
+    }
+
+    return this;
+  }
+
+  protected PopupPanel getRootPopup() {
+    Widget parent = getRootMenu().getParent();
+    while ( parent != null ) {
+      if ( parent instanceof PopupPanel ) {
+        return (BurgerMenuPopup) parent;
+      }
+
+      parent = parent.getParent();
+    }
+
+    return null;
+  }
+
+  // Local version of private super.findItem( Element ).
+  protected MenuItem findItemLocal( com.google.gwt.dom.client.Element hItem ) {
+    for ( MenuItem item : getItems() ) {
+      if ( item.getElement().isOrHasChild( hItem ) ) {
+        return item;
+      }
+    }
+
+    return null;
+  }
+
+  // Local version of private super.eatEvent( Event ).
+  protected void eatEventLocal( Event event ) {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  // Local version of private super.close( boolean ).
+  protected void closeLocal( boolean focus ) {
+    BurgerMenuBar parentMenuBar = getMenuBarParent();
+    if ( parentMenuBar != null ) {
+      PopupPanel popupPanel = parentMenuBar.getMenuBarPopup();
+      if ( popupPanel != null ) {
+        popupPanel.hide( !focus );
+      }
+
+      if ( focus ) {
+        parentMenuBar.focus();
+      }
+    }
+  }
+
+  // Local version of private super.selectFirstItemIfNoneSelected().
+  private boolean selectFirstItemIfNoneSelectedLocal() {
+    if ( getSelectedItem() == null ) {
+      for ( MenuItem nextItem : getItems() ) {
+        if ( nextItem.isEnabled() ) {
+          selectItem( nextItem );
+          break;
+        }
+      }
+
+      return true;
+    }
+
+    return false;
+  }
+
+  // Local, adapted version of private super.doItemAction(MenuItem, boolean, boolean).
+  // Overrides the command execution of a menu item closes the whole menu (unlike closeAllParents()).
+  // Additionally, the special back menu items do not cause the menu to close.
+  protected void doItemActionLocal( MenuItem item, boolean fireCommand, boolean focus ) {
```

Comment:
Is the function for MenuBar#doItemAction never used in this context?
Can you just add the annotation `@Override` a function `doItemAction` and use your custom implementation?